gateway/upstreamer/push/upstreamer.go

```diff
@@ -226,3 +245,30 @@ func (c *Upstreamer) listenService(ctx context.Context, ready chan struct{}) {
 		}
 	}
 }
+
+// CollectLatency implement the LatencyBasedUpstreamer interface to add new
+// samples into the latencies sync map
+func (c *Upstreamer) CollectLatency(address string, responseTime time.Duration) {
+
+	if values, ok := c.latencies.Load(address); ok {
+		values.(*movingAverage).insertValue(float64(responseTime.Microseconds()))
+	} else {
+		c.latencies.Store(address, newMovingAverage(c.config.latencySampleSize))
+		c.CollectLatency(address, responseTime)
+	}
+}
+
+// delete remove the latency tracking for a give address
+func (c *Upstreamer) delete(address string) {
```

Comment:
What's the use for this function?

Reply:
removing and outdated service from the sync map.